packages/imperative/src/auth/handlers/BaseAuthHandler.ts

```diff
@@ -197,29 +196,37 @@ export abstract class BaseAuthHandler extends AbstractAuthHandler {
      * @param {IHandlerParameters} params Command parameters sent by imperative.
      */
     protected async processLogout(params: IHandlerParameters) {
-        ImperativeExpect.toNotBeNullOrUndefined(params.arguments.tokenValue, "Token value not supplied, but is required for logout.");
-
-        // Force to use of token value, in case user and/or password also on base profile, make user undefined.
-        if (params.arguments.user != null) {
+        // Force the use of token value, in case user and/or password are also provided.
+        if (params.arguments.tokenValue != null &&
+            (params.arguments.user != null || params.arguments.password != null)) {
             params.arguments.user = undefined;
+            params.arguments.password = undefined;
         }
 
-        params.arguments.tokenType = this.mDefaultTokenType;
+        if (params.arguments.tokenType == null) {
+            params.arguments.tokenType = this.mDefaultTokenType;
+        }
```

Comment:
Should we handle removing certificates from the session as well? I guess this may not be a concern, if tokens have a higher priority than certificates when both are present 😅